src/event_pusher/mod_event_pusher_push.erl

```diff
@@ -79,6 +84,7 @@ start(Host, Opts) ->
     gen_iq_handler:add_iq_handler(ejabberd_sm, Host, ?NS_PUSH, ?MODULE,
                                   iq_handler, IQDisc),
 
+    ejabberd_hooks:add(unacknowledged_message, Host, ?MODULE, unacknowledged_message, 90),
```

Comment:
I intentionally bypassed event pusher module. don't think that other pusher backends are interested in this hook.

Reply:
They might be. It breaks the consistency in `mod_event_pusher`.

Reply:
I agree, other modules could be interested in that, but...
This event has slightly different semantic. Existing hooks generate events "hey, there is a new message, please check if you want to push it and maybe do so". This hook, generate "key, there is a message that could be delivered, you SHOULD push it". But still, IMHO, this could be the same path, only with additional props that inform the push module what is the trigger reason, so that the custom, installation-specific function `should_publish` could make an inform decision about pushing the notification in both cases. There are clients that user push notification with "master" switch that disables ALL notifications in the system, and this behaviour is implemented in mentioned  `should_publish` plugin function. Bypassing that breaks such feature and renders `should_publish` _plugin_ function, that is intended for customisation useless. 

Therefore, my proposition is as follows:
Let's stick to `mod_event_pusher_hook_translator` and just refactor it a bit by somehow adding notification trigger reason. Based on that, `should_publish` could either check if user is online (for current triggers) or skip the check if the trigger was by un-acked message. 

What do you think about it @DenysGonchar @fenek ?

Reply:
I agree, it's a good idea to reuse existing record and add `reason` or `condition` field.

Reply:
I wish to resolve merge conflicts and other comments, but can we make this one in another PR?